### PR TITLE
Survey scanner

### PR DESCRIPTION
**Pull request by Erikafox**
## About The Pull Request
Adds survey scanners and survey points. 
A survey scanner is a backpack slot piece of equipment that holds a survey scanner, and allows you to scan a survey point.
Scanning a survey point creates science points, like the survey scanner of ole.
https://github.com/shiptest-ss13/Shiptest/assets/94164348/848f3fe3-af38-4a7c-b5c9-bd14a9f9ad8b

Dissections have been fairly heavily nerfed to account for this, and to make surveying a desirable thing to do.

All sprites are placeholder.

Also adds a framework for other tools that require a backpack to use.

## Why It's Good For The Game
Less Dissections, More Exploration

## Changelog

:cl:
add: Survey Backpacks! You can buy them at the outpost and use them to survey new survey points on planets
del: Survey Handhelds have been nuked
balance: Dissection science has been decreased across the board

/:cl:

### CODE REVIEW

File: code/game/objects/items/survery_handheld.dm [hunk 1]
```diff
@@ -1,136 +1,169 @@
...
 
-/obj/item/survey_handheld/attack_self(mob/user)
-	if(active)
+
+/obj/item/attachment/survey_scanner
+	name = "survey scanner"
+	desc = "A wired tool designed to work in tandem with a survey pack"
+	w_class = WEIGHT_CLASS_BULKY
+	icon = 'icons/obj/item/survey_handheld.dmi'
+	icon_state = "survey_att"
```
Comment (thgvr) [inline code/game/objects/items/survery_handheld.dm]: survey-att is the iconstate in the dmi

Comment (Erikafox) [reply code/game/objects/items/survery_handheld.dm]: balls

**Comment by thgvr** [comment]
oooo you want to finish this

**Comment by Erikafox** [comment]
> oooo you want to finish this

The only thing this needs is a proper overlay and I couldn't figure that out over a week. If you or someone else can help me with it I am more than happy to finish it up.

File: code/modules/clothing/clothing.dm [hunk 2]
```diff
@@ -282,6 +282,7 @@
 * SEE_TURFS // can see all turfs (and areas), no matter what
 * SEE_PIXELS// if an object is located on an unlit area, but some of its pixels are
 *           // in a lit area (via pixel_x,y or smooth movement), can see those pixels
+* SEE_SCIENCE // can see science points
```
Comment (ZephyrTFA) [inline code/modules/clothing/clothing.dm]: That's not how this works at all.
These are  BYOND defines you cannot extend this functionality.
https://www.byond.com/docs/ref/#/mob/var/sight

Comment (Erikafox) [reply code/modules/clothing/clothing.dm]: oh that's leftover.

File: code/modules/cargo/packs/exploration.dm [hunk 3]
```diff
@@ -89,3 +89,22 @@
 		/obj/item/survivalcapsule,
 	)
 
+/datum/supply_pack/exploration/scanners
+	name = "Survey Scanner Kit"
+	desc = "Contains a Survey Scanner and survey locator, ideal for analyzing the surface of planets."
+	cost = 1250
```
Comment (thgvr) [inline code/modules/cargo/packs/exploration.dm]:
```suggestion
	cost = 1000
```

File: code/modules/cargo/packs/exploration.dm [hunk 4]
```diff
@@ -89,3 +89,22 @@
...
+	contains = list(
+		/obj/item/gear_pack/survey_pack,
+		/obj/item/pinpointer/survey_data
+	)
+	crate_name = "Survey Scanner Kit"
+
+/datum/supply_pack/exploration/adv_scanner
+	name = "Advanced Survey Scanner Kit"
+	desc = "Contains a state of the art Survey Scanner"
+	cost = 2000
```
Comment (thgvr) [inline code/modules/cargo/packs/exploration.dm]:
```suggestion
	cost = 1500
```

File: code/game/objects/items/survery_handheld.dm [hunk 5]
```diff
@@ -1,136 +1,124 @@
...
+	slowdown = 1
+	var/survey_mult = 1
 	var/survey_delay = 4 SECONDS
 
-/obj/item/survey_handheld/advanced
-	name = "Advanced Survey Handheld"
-	desc = "An improved version of its predecessor this tool collects large amounts of data."
+/obj/item/gear_pack/survey_pack/advanced //can be purchased, is Expendy.
+	name = "Advanced Survey Pack"
+	desc = "A high hech piece of scientific kit designed for thorough planetary survey"
```
Comment (thgvr) [inline code/game/objects/items/survery_handheld.dm]:
```suggestion
	desc = "A high tech piece of scientific kit designed for thorough planetary survey."
```

File: code/game/objects/items/survery_handheld.dm [hunk 6]
```diff
@@ -1,136 +1,124 @@
-/obj/item/survey_handheld
-	name = "Survey Handheld"
-	desc = "A small tool designed for quick and inefficient data collection about your local star sector."
+/obj/item/gear_pack/survey_pack
+	name = "Survey Pack"
+	desc = "A large scientific kit designed for planetary survey"
```
Comment (thgvr) [inline code/game/objects/items/survery_handheld.dm]:
```suggestion
	desc = "A large scientific kit designed for planetary survey."
```

File: code/game/objects/items/gear_packs.dm [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+		snap_back()
+
+/obj/item/attachment/dropped(mob/user)
+	. = ..()
+	if(!req_pack)
+		return ..()
+	if(user)
+		UnregisterSignal(user, COMSIG_MOVABLE_MOVED)
+		if(user != loc)
+			to_chat(user, "<span class='notice'>[src] snap back into the main unit.</span>")
```
Comment (thgvr) [inline code/game/objects/items/gear_packs.dm:275]:
```suggestion
			to_chat(user, "<span class='notice'>[src] snaps back into the main unit.</span>")
```

File: code/game/objects/items/gear_packs.dm [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+	SIGNAL_HANDLER
+
+	if(!req_pack ||!pack)
+		return
+	if(!in_range(src,pack))
+		var/mob/living/L = loc
+		if(istype(L))
+			to_chat(L, "<span class='warning'>[pack]'s [src] overextends and come out of your hands!</span>")
+		else
+			visible_message("<span class='notice'>[src] snap back into [pack].</span>")
```
Comment (thgvr) [inline code/game/objects/items/gear_packs.dm]:
```suggestion
			visible_message("<span class='notice'>[src] snaps back into [pack].</span>")
```

File: code/game/objects/items/gear_packs.dm [hunk 9]
```diff
@@ -0,0 +1,285 @@
...
+
+/obj/item/attachment/proc/check_range()
+	SIGNAL_HANDLER
+
+	if(!req_pack ||!pack)
+		return
+	if(!in_range(src,pack))
+		var/mob/living/L = loc
+		if(istype(L))
+			to_chat(L, "<span class='warning'>[pack]'s [src] overextends and come out of your hands!</span>")
```
Comment (thgvr) [inline code/game/objects/items/gear_packs.dm]:
```suggestion
			to_chat(L, "<span class='warning'>[pack]'s [src] overextends and comes out of your hands!</span>")
```

**Review by thgvr — CHANGES_REQUESTED** [review]
nitpicks

**Pull request closed without merging**